poem/src/web/mod.rs

```diff
@@ -326,6 +328,16 @@ pub trait FromRequest<'a>: Sized {
     async fn from_request_without_body(req: &'a Request) -> Result<Self> {
         Self::from_request(req, &mut Default::default()).await
     }
+
+    /// Extract from request head and body synchronously.
+    fn from_request_sync(_req: &'a Request, _body: &mut RequestBody) -> Result<Self> {
+        panic!("Not implemented, please implement one of from_request and from_request_sync");
+    }
+
+    /// Extract from request head synchronously.
+    fn from_request_without_body_sync(req: &'a Request) -> Result<Self> {
+        Self::from_request_sync(req, &mut Default::default())
+    }
```

Comment:
I don't think it's necessary, what do you think?

```suggestion

    /// Extract from request head and body synchronously.
    fn from_request_sync(_req: &'a Request, _body: &mut RequestBody) -> Result<Self> {
        panic!("Not implemented, please implement one of from_request and from_request_sync");
    }

    /// Extract from request head synchronously.
    fn from_request_without_body_sync(req: &'a Request) -> Result<Self> {
        Self::from_request_sync(req, &mut Default::default())
    }
```